src/components/Application/ProgressBarNEW.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import { Accordion, Container, Grid, Header, Icon, Label, List, Sticky } from 'semantic-ui-react'
+import strings from '../../utils/constants'
+import { FullStructure, PageNEW, Progress } from '../../utils/types'
+
+interface ProgressBarProps {
+  structure: FullStructure
+  changePage: (sectionCode: string, pageNumber: number) => void
+  current: {
+    sectionCode: string
+    page: number
+  }
+}
+
+const ProgressBarNEW: React.FC<ProgressBarProps> = ({ structure, changePage, current }) => {
+  const {
+    info: { isLinear },
+    sections,
+  } = structure
+
+  const activeIndex =
+    Object.values(sections).findIndex(({ details }) => details.code === current.sectionCode) || 0
+
+  const getPageList = (sectionCode: string, pages: { [pageName: string]: PageNEW }) => {
+    const isActivePage = (sectionCode: string, pageNumber: number) =>
+      current.sectionCode === sectionCode && current.page === pageNumber
+
+    return (
+      <List
+        link
+        style={{ paddingLeft: '50px' }}
+        items={Object.entries(pages).map(([pageName, { number, progress }]) => ({
+          key: `ProgressSection_${sectionCode}_${number}`,
+          active: isActivePage(sectionCode, number),
+          as: 'a',
+          icon: progress ? getIndicator(progress) : null,
+          content: pageName,
+          onClick: () => changePage(sectionCode, number),
+        }))}
+      />
+    )
+  }
+
+  const getIndicator = (progress: Progress, step?: number) => {
+    const { completed, valid } = progress
+    if (!completed)
```

Comment:
Hmm, don't we want to show error if it's not completed but something is defiantly invalid ?

Reply:
Ohh looks like it does wen page refresh, never mind